src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -99,15 +99,20 @@ def create_webapp(cmd, resource_group_name, name, plan, runtime=None, startup_fi
         raise MutuallyExclusiveArgumentError('usage error: --deployment-source-url <url> | --deployment-local-git')
 
     docker_registry_server_url = parse_docker_image_name(deployment_container_image_name)
-
+    
     client = web_client_factory(cmd.cli_ctx)
+    plan_info = None
     if is_valid_resource_id(plan):
         parse_result = parse_resource_id(plan)
         plan_info = client.app_service_plans.get(parse_result['resource_group'], parse_result['name'])
     else:
-        plan_info = client.app_service_plans.get(name=plan, resource_group_name=resource_group_name)
+        plans = list(client.app_service_plans.list(detailed=True))
+        for user_plan in plans: 
+            if user_plan.name.lower() == plan.lower():
+                parse_result = parse_resource_id(user_plan.id)
+                plan_info = client.app_service_plans.get(parse_result['resource_group'], parse_result['name'])
```

Comment:
It looks this logic selects the plan to use by listing the plans in the user's subscription and then getting the first one that matches the name the user passed in. 

We could do it this way, or we could require the user to specify a resource ID if the plan is in a different RG. I like that the second is a little more deterministic (doesn't rely on the ordering of the list API) -- @panchagnula do you have thoughts on which approach would be better here? 

To be fair, I've seen both approaches done in the CLI (the first for SWAs and the second for vnets).

Reply:
Actually, I just checked this and it looks like we are already allowing the user to pass in a plan in a different RG by RID. So this point is not relevant. 

We should still have it default the plan in the web app's RG though (see my previous comment)